lib/AST/Evaluator.cpp

```diff
@@ -381,8 +371,7 @@ void Evaluator::dumpDependenciesGraphviz() const {
 
 void evaluator::DependencyRecorder::realize(
     const DependencyCollector::Reference &ref) {
-  auto *source = getActiveDependencySourceOrNull();
-  assert(source && "cannot realize dependency without associated file!");
```

Comment:
Is it worth keeping this assertion?

Reply:
I removed it since `NullablePtr::get` is going to fire its own assertion if we get this wrong at some point. I could leave it in so we get more diversified stack traces, but since it's an internal invariant if it fails something catastrophic really has gone wrong.

Reply:
Makes sense. Thanks for explaining.